### PR TITLE
Nfdiv 1568 refactor awaiting conditional order notification

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1568

### Change description ###
Move isDivorce method inside CaseData
Refactor conditional order notifications to use NotificiationDispatcher

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/common/notification/ConditionalOrderPendingReminderNotification.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                CITIZEN_APPLY_FOR_CONDITIONAL_ORDER,
+                templateVars,
+                applicant1.getLanguagePreference()
+            );
+        }
+    }
+
+    @Override
+    public void sendToApplicant1Solicitor(final CaseData caseData, final Long id) {
+        if (isNull(caseData.getConditionalOrder().getConditionalOrderApplicant1Questions().getSubmittedDate())) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/notification/ConditionalOrderPendingReminderNotification.java:54]: you can create a method in conditional order something like isConditionalOrderPending()...

File: src/main/java/uk/gov/hmcts/divorce/common/notification/ConditionalOrderPendingReminderNotification.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                templateVars,
+                applicant.getLanguagePreference()
+            );
+            log.info("Successfully sent conditional order pending notification for case : {}", id);
+        }
+    }
+
+    @Override
+    public void sendToApplicant2(final CaseData caseData, final Long id) {
+        if (!isNull(caseData.getConditionalOrder().getConditionalOrderApplicant1Questions().getSubmittedDate())
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/notification/ConditionalOrderPendingReminderNotification.java:72]: can reuse the method here `isConditionalOrderPending`

File: src/main/java/uk/gov/hmcts/divorce/common/notification/AwaitingConditionalOrderNotification.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        final Map<String, String> templateVars = commonContent.basicTemplateVars(caseData, id);
+        templateVars.put(SOLICITOR_NAME, applicant1.getSolicitor().getName());
+
+        notificationService.sendEmail(
+            applicant1.getSolicitor().getEmail(),
+            SOLICITOR_AWAITING_CONDITIONAL_ORDER,
+            templateVars,
+            applicant1.getLanguagePreference()
+        );
+        log.info("Successfully sent awaiting conditional order notification for case : {}", id);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/common/notification/AwaitingConditionalOrderNotification.java:60]: this last log statement is present in some methods but not in all so would be nice to make it consistent

**Review by nitinprabhu — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/common/notification/AwaitingConditionalOrderNotification.java (hunk 3 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/common/notification/AwaitingConditionalOrderNotification.java:60]: looks like one I forgot to remove, I will change

File: src/main/java/uk/gov/hmcts/divorce/common/notification/ConditionalOrderPendingReminderNotification.java (hunk 1 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/common/notification/ConditionalOrderPendingReminderNotification.java]: Update in https://github.com/hmcts/nfdiv-case-api/pull/1165/commits/7e8d908c1b49cd31b78ec85a866e0f3fa7e10a41

**Pull request merged**